### PR TITLE
Add RandomFloat4D node

### PR DESCRIPTION
**Pull request by AlexeyBond**
This adds a random noise function I previously added to https://github.com/Maujoe/Godot-Visual-Shader-Node-Library (which wasn't ported to Godot 4).

A random function with more than 2 inputs makes it possible to generate a 2D noise changing over time - that's what I tried to do recently using this addon but couldn't find an appropriate node.

### CODE REVIEW

**Comment by arkology** [comment]
Thanks a lot for your contribution! 
I'll try to review (I'm sure everything is fine, but still) and merge at nearest weekends. 

Also it would be great if you will create another PR to backport changes to 3.x branch (https://github.com/arkology/ShaderV/tree/godot-3.x) after merging this PR. Or I could backport if by myself, it's not a problem.

File: addons/shaderV/tools/random/randomFloat4D.gd [hunk 1]
```diff
@@ -57,5 +55,13 @@ func _get_global_code(mode):
 	return '#include "' + path + '/randomFloat4D.gdshaderinc"'
 
 func _get_code(input_vars, output_vars, mode, type):
+	var input = input_vars[0]
+	var offset = input_vars[2]
+
+	if not input:
+		input = 'vec4(0)'
+	if not offset:
+		offset = 'vec4(0)'
```
Comment (AlexeyBond) [inline addons/shaderV/tools/random/randomFloat4D.gd:64]: 🤦 Godot doesn't handle default values on Vector4 inputs

https://github.com/godotengine/godot/pull/90850

**Review by arkology — CHANGES_REQUESTED** [review]
ATM I get following errors when disconnecting `input` or `offset` inputs of RandomFloat4D node in basic_examples scene:
```
  res://addons/shaderV/examples/basic_examples.tscn::VisualShader_y00xk:19 - Invalid arguments to operator '+': 'vec3, vec4'.
  Shader compilation failed.
--res://addons/shaderV/examples/basic_examples.tscn::VisualShader_y00xk--
    1 | shader_type canvas_item;
    2 | render_mode blend_mix;
    3 | 
    4 | 
    5 | 
    6 | 
    7 | #include "res://addons/shaderV/tools/random/randomFloat4D.gdshaderinc"
    8 | 
    9 | 
   10 | void fragment() {
   11 | 
   12 |  float n_out8p0 = TIME;
   13 | 
   14 | 
   15 |  float n_out9p0;
   16 | 
   17 |  vec3 n_in9p0 = vec3(0.00000, 0.00000, 0.00000);
   18 |  float n_in9p1 = 1.00000;
E  19->  {
   20 |  n_out9p0 = _randFloat4D(n_in9p0 * n_in9p1 + vec4(n_out8p0));
   21 |  }
   22 | 
   23 | 
   24 | 
   25 |  COLOR.rgb = vec3(n_out9p0);
   26 | 
   27 | 
   28 | }
   29 | 
--res://addons/shaderV/tools/random/randomFloat4D.gdshaderinc--
    1 | highp float _randFloat4D(vec4 input) {
    2 |  float f = dot(fract(input) + fract(input * 2.32184321231),vec4(129.898,782.33,944.32214932,122.2834234542));
    3 |  return fract(sin(f) * 437588.5453);
    4 | }
    5 | 
```

File: addons/shaderV/examples/basic_examples.tscn [hunk 2]
```diff
@@ -1592,7 +1592,7 @@ input_name = "uv"
 input_name = "time"
 
 [sub_resource type="VisualShaderNodeCustom" id="VisualShaderNodeCustom_swrn3"]
-default_input_values = [0, Vector3(0, 0, 0), 1, 1.0, 2, Vector3(0, 0, 0)]
```
Comment (AlexeyBond) [inline addons/shaderV/examples/basic_examples.tscn]: There were invalid default values from previous iteration stored in the scene. It should be fine now.

**Review by arkology — APPROVED** [review]

**Comment by arkology** [comment]
Thanks!

**Pull request merged**